unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2597,30 +2603,39 @@ protected void updateAllAgentCollidersForVisibilityCheck(bool enableColliders)
         }
 
         public SimObjPhysics[] VisibleSimObjs(string objectId, bool forceVisible = false) {
-            ServerAction action = new ServerAction();
-            action.objectId = objectId;
-            action.forceVisible = forceVisible;
-            return VisibleSimObjs(action);
+            SimObjPhysics sop = getSimObjectFromTypeOrId(objectType: null, objectId: objectId);
+            List<SimObjPhysics> simObjs = new List<SimObjPhysics>();
+            if (sop != null) {
+                simObjs.Add(sop);
+            }
+            if (forceVisible) {
+                return simObjs.ToArray();
+            } else {
+                return GetAllVisibleSimObjPhysics(m_Camera, maxVisibleDistance, simObjs);
+            }
         }
 
 
         public SimObjPhysics[] VisibleSimObjs(ServerAction action) {
             List<SimObjPhysics> simObjs = new List<SimObjPhysics>();
+            SimObjPhysics[] visibleSimObjs = null;
+            if (!string.IsNullOrEmpty(action.objectId))
+            {
+                // if an objectId was passed in we can limit the check to just the target object
+                // in theory we could return from here, but its possible that a conflicting 
+                // objectType is passed in which would then filter out a found object
```

Comment:
I would vote for removing this behavior. People should instead call the hypothetical `GetSimObjFromId(objectId, excludeIfNotVisible)` function if they want this.